<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <title>Internet Resources for Molecular Biologists</title>
    <link rel="stylesheet" href="style.css" />

    <!-- Google Custom Search Engine -->
    <script
      async
      src="https://cse.google.com/cse.js?cx=5306da9863c9d4c77">
    </script>
  </head>



  <body>
    <header>

      <div class="logo">
        <a href="index.html">Resource Hub</a>
      </div>

      <nav>

        <ul>
          <li><a href="index.html">Home</a></li>
          <li><a href="index.html#categories">Categories</a></li>
          <li><a href="aboutUs.html">About Us</a></li>
        </ul>

      </nav>

      <div class="search-container">
        <div class="gcse-search"></div>
      </div>

    </header>


    <div class="breadcrumbs">
      <a href="index.html">Home</a> ›
      <a href="index.html#categories">Categories</a> ›

      <!--Current Breadcrumb-->
      <span>Metabolomics</span>

    </div>


    <main class="content">

      <!--Title-->
      <h1>Metabolomics</h1>



      <p>

        A good place to start would be the

        <a href="https://metabolomicscentre.ca/" target="_blank">

          Metabolomics Innovation Centre

        </a>

        at the University of Alberta and their list of software.

      </p>



      <section class="tutorial-section">

        <ul>

          <li>

            <a href="https://bmrb.io/" target="_blank">

              Biological Magnetic Resonance Data Bank Metabolomics

            </a>

            - is a repository for data from NMR spectroscopy on proteins,

            peptides, nucleic acids, and other biomolecules.

          </li>

          <br />

          <li>

            <a href="#" target="_blank">COLMAR</a>

            Complex Mixture Analysis by NMR (OSU, Campus Chemical Instrument

            Center) has a number of tools: Covariance: Generates covariance NMR

            spectrum from TOCSY-type experiments; DemixC: Deconvolutes 2D TOCSY

            spectra of complex mixtures into their individual components; 1D

            NMR: Query 1D 1H and 1D 13C NMR spectra; 13C-TOCCATA: Query 2D

            13C-13C TOCSY spectra; 1H(13C)-TOCCATA: Query 2D 1H-1H TOCSY and 2D

            13C-1H HSQC-TOCSY spectra; 13C-1H HSQC: Query 2D 13C-1H HSQC

            spectra; COLMARm: Query Multiple 2D spectra (HSQC, TOCSY and

            HSQC-TOCSY.)

          </li>

          <br />

          <li>

            <a href="https://ecmdb.ca/" target="_blank">ECMDB</a>

            E.coli Metabolome Database - is an expertly curated database

            containing extensive metabolomic data and metabolic pathway diagrams

            about Escherichia coli (strain K12, MG1655). This database includes

            significant quantities of "original" data compiled by members of

            the Wishart laboratory as well as additional material derived from

            hundreds of textbooks, scientific journals, metabolic

            reconstructions and other electronic databases. ECMDB currently

            contains 3755 small molecules with 1402 associated enzymes and 387

            associated transporters. It also has 1542 metabolic pathways that

            are linked to 3011 metabolites. A total of 19,294 NMR and MS

            spectra (experimental and predicted) for 3098 different E. coli

            metabolites are also contained in the database.

            <br />

            (

            <strong>Reference:</strong>

            T. Sajed et al. (2016). Nucleic Acids Res, 44(D1):D495-501).

          </li>

          <br />

          <li>

            <a

              href="https://mips.helmholtz-muenchen.de/proj/GWAS/gwas/index.php"

              target="_blank">

              Metabolomics GWAS Server

            </a>

            . This site contains the association results of our two genome-wide

            association studies on the human metabolome. The server is currently

            under development and new features will be added as soon as they

            are available. Especially we will add more parameters for a more

            convenient browsing of the data.

          </li>

          <br />

          <li>

            <a href="https://m2ia.met-bioinformatics.cn/" target="_blank">

              M<sup>2</sup>IA

            </a>

            is a web-based server for the Microbiome and Metabolome data

            integration analysis. M<sup>2</sup>IA integrates a variety of

            statistical analysis methods, including univariate analysis,

            multivariate modeling, and functional network analysis. Users can

            upload data, customize parameters, and submit a job by one click.

            An html-based summarized report will be generated automatically for

            overview, accompanied by supporting tables and high-resolution

            figures available for downloading.

          </li>

          <br />

          <li>

            <a href="https://www.metaboanalyst.ca/" target="_blank">

              MetaboAnalyst

            </a>

            - is a web server designed to permit comprehensive metabolomic data

            analysis, visualization and interpretation. It supports a wide

            range of complex statistical calculations and high quality

            graphical rendering functions that require significant

            computational resources.

            <br />

            (

            <strong>Reference:</strong>

            J.Xia et al. 2015. Nucl. Acids Res. 43 (W1): W251-W257).

          </li>

          <br />

          <li>

            <a href="https://workflow4metabolomics.org/" target="_blank">

              Workflow4Metabolomics

            </a>

            - is the first fully open-source and collaborative online platform

            for computational metabolomics. W4M is a virtual research

            environment built upon the Galaxy web-based platform technology. It

            enables ergonomic integration, exchange and running of individual

            modules and workflows. Alternatively, the whole W4M framework and

            computational tools can be downloaded as a virtual machine for

            local installation.

            <br />

            (

            <strong>Reference:</strong>

            Giacomoni F et al. 2015. Bioinformatics 31:1493-1495).

          </li>

          <br />

        </ul>

      </section>



      <p class="updated"><strong>Updated:</strong> November, 2025</p>

    </main>



    <footer>

      <p>© 2025 Internet Resources for Molecular Biologists</p>

    </footer>

  </body>

</html>

