<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Buffers - Resource Hub</title>
  <link rel="stylesheet" href="../style.css">
</head>
<body>
  <header>
    <div class="logo"><a href="../index.html">Resource Hub</a></div>
    <nav>
      <ul>
        <li><a href="../index.html">Home</a></li>
        <li><a href="../index.html#categories">Categories</a></li>
        <li><a href="../aboutUs.html">About Us</a></li>
      </ul>
    </nav>
    <div class="search-container">
      <input type="text" id="search-bar" placeholder="Search categories..." onkeyup="searchSite()">
      <div id="search-results" class="search-results"></div>
    </div>
  </header>

  <div class="breadcrumbs">
    <a href="../index.html">Home</a> ›
    <a href="../index.html#categories">Categories</a> ›
    <a href="miscellaneous.html">Miscellaneous</a> › 
    <span>Buffers</span>
    
  </div>

  <main class="content">
    <h1>Buffers</h1>


    <section class="tutorial-section">
    <section class="tool-section">

    <ul>
      <li><a href="http://www.cellbiol.com/scripts/cleaner/dna_protein_sequence_cleaner.php">CLEANER</a> (cellbiol.com) Offers and useful alternative by stripping off nonsequence material (e.g. numbers) from nucleic acid and protein sequence.  N.B. Particularly useful if you only want part of the DNA sequence from GenBank. Other sites offering a program to convert sequence format are <a href="http://phylogeny.lirmm.fr/phylo_cgi/one_task.cgi?task_type=readseq">Readseq</a> (D.G. Gilbert, Univ. Indiana, U.S.A.). See also <a href="#">CONVERT</a></li>
      <br>

      <li><a href="https://reverse-complement.com/cleanup.html">Sequence Cleanup or Formatting</a> - offers: 60 characters per line, UPPER CASE, lower case, GenBank, and skyline</li>
      <br>

      <li>
        <a href="http://seqscrub.gabefoley.com/">SeqScrub</a> is a web application that cleans up FASTA file headers and appends information from external databases 
        <br>(<strong>Reference:</strong> Foley G et al. (2019)  Biotechniques. 67(2) :50-54).
      </li>
      <br>

      <li><a href="https://biomodel.uah.es/en/lab/cybertory/analysis/massager.htm">Nucleic Acid Sequence Massager</a> (formerly at Allotron Biosensor Corporation) - my favourite which in addition to removing spurious material (numbers, breaks, HTML, spaces) changes the format (upper to low case, complement, reverse, RNA to DNA, and triplets).  It is also available <a href="https://www.cmbn.no/tonjum/seqMassager-saf.htm">here.</a></li>
      <br>

      <li><a href="http://www.mathaddict.net/dnatranslate3.htm">DNA Sequence Reverse and Complement </a> (cellbiol.com) - With this tool you can reverse a DNA sequence, complement a DNA sequence or reverse and complement a DNA sequence. It also removes numbers, spaces, tabs, slash or non-GATC characters; and,  allows conversion to upper or low case.</li>
      <br>

      <li><a href="https://biomodel.uah.es/en/lab/cybertory/analysis/secret.htm">Secret encoding tool</a> to encode secret messages in DNA sequence. Alternatively use <a href="https://www.thinkbiotech.com/DNA-o-gram/encode.php">DNA-0-gram Generator</a> which allows one to  Send a DNA-o-gram or Decode a DNA-o-gram.</li>
      <br>

      <li>Radioactive decay: see <a href="https://www.1728.org/decayexp.htm">Radioactive Decay Calculator</a> (1728 Software Systems) or <a href="http://www.convertalot.com/decaycalculator.html">Radioactive Decay Calculator 2.</a></li>
      <br>

      <li><a href="https://daleked.pages.iu.edu/centhelp.html">Centrifuge Helper</a> (formerly D.Wang, Oklahoma State University, U.S.A.) provides one with maximum RPM, RCF (ave, min, max), K factor and comparison between two rotors (Sorvall, Beckman, Composite, Eppendorf, IEC and Piramoon) or <a href="https://www.sciencegateway.org/tools/rotor.htm">Centrifuge Rotor Speed Calculator</a> (Science Gateway).</li>
      <br>

      <li><a href="https://www.easyunitconverter.com/">Unit Converter</a> - allows you to check all type of metric conversion of measurement units example: length, weight, height, area, temperature, time, pressure, speed, force, volume, data storage, fuel consumption, energy, degree, number converter, etc and also you can check online calculators basic & advanced math calculations like percentage, mortgage, fraction, interest rate, loan, tax, GST, tip, love, work, algebra, finance calculator and health calculator, etc.. </li>
      <br>

      <li>
        <a href="#">Calculations</a> (Zbio.net, Russia) Spectrophotometric measurement of nucleic acids concentration; Conversion: speed of rotation [krpm] - Relative Centrifuge Force [kg];
        <ul>
            <li>Conversion: radioactivity - biological amount of dNTP's;</li>
            <li>Conversion: weight - moles (for proteins).</li>
            <li>Conversion: weight - moles (for nucleic acids).</li>
            <li>Conversion: DNA - protein.</li>
            <li>Conversion: molar quantity - molar concentration.</li>
            <li>Conversion: units of measurement. </li>
            <li>Estimation of  time for bacteria growth.</li>
        </ul>
      </li>
      <br>

      <li>
        <a href="#">Protocols</a> (Zbio.net, Russia) - provides an incredible set of data and calculators for:
        <ul>
            <li><a href="#">Densities of liquids</a></li>
            <li><a href="#">Densities of salt solutions</a></li>
            <li><a href="#">Densities of acids, alkali and organic compounds</a></li>
            <li><a href="#">Salt solutions (part one)</a></li>
            <li><a href="#">Salt solutions (part two)</a></li>
        </ul>
      </li>
      <br>

      <li><a href="https://www.promega.ca/resources/tools/biomath/">BioMath Calculators</a> (Promega Corp., U.S.A.) - DNA and protein unit conversions; temperature (°C, °F  & °K); and, Tm calculations for oligonucleotides, includes adjustments for Mg+2 concentration. Highly recommended.</li>
      <br>

      <li><a href="https://goodcalculators.com/">Good Calculators</a> - Free Online Calculators - This website provides a variety of online calculators, math and physics, engineering and conversion calculators.</li>
      <br>

      <li><a href="http://www.kenkyuu.net/js/nacalc.html">Nucleic acid calculator</a> (Kenkyuu-Ryuugaku NET) or <a href="https://mcb.berkeley.edu/labs/krantz/tools/oligocalc.html">Oligo Calc: Oligonucleotide Properties Calculator</a> - From A260nm calculate DNA, RNA and oligonucleotide concentrations online. Similar site is<a href="https://oligocalc.eu/">here</a></li>
      <br>

      <li><a href="https://www2.iib.uam.es/bioinfo/packages/jambw/5/3/mapcalc3.html">Physical mapping calculator - library construction</a> (Andrei Grigoriev) - shows diagrammatically how the clone size and number influence what percentage of the genome is covered.</li>
      <br>

      <li><a href="https://phbuffers.org/BuffferCalc/Buffer.html">Biological buffer calculator</a> (R. Beynon, Dept. Veterinary Preclinical Sciences Univ. Liverpool, England) - controls for concentration, ionic strength and temperature. Also available <a href="https://www.bioinformatics.org/jambw/5/4/index.html">here</a>. For pKa data see <a href="https://www.sigmaaldrich.com/CA/en/support/calculators-and-apps/buffer-calculator">here</a> (Sigma-Aldrich).  Other buffer sites include: <a href="https://www.sciencegateway.org/tools/buffer.htm">Science Gateway Buffer Calculator</a>, and <a href="#">Buffer Calculator</a></li>
      <br>

      <li><a href="https://restrictionmapper.org/dilutioncalc9.htm">Dilution Calculator</a> (www.restrictionmapper.org/) - this very useful site for the mathematically challenged provides percent by weight, weight per volume and molarity of solution from stock solutions or solids.</li>
      <br>
      
    </ul>
  </section>



  <section class="tool-section">
    <h2>Sites which permit one to carry out a variety of analyses include:</h2>

    <ul>
      <li>EMBOSS (European Molecular Biology Open Source  Software Suite) - can be accessed <a href="https://emboss.sourceforge.net/">here</a> (SourceForge, U.S.A.).  A list of their programs is <a href="https://molbiol-tools.ca/EMBOSS.html">appended</a>.  Some of threse tools can be accessed at the <a href="https://www.ebi.ac.uk/jdispatcher/">EMBL-EBI site.</a></li>
      <br>

      <li><a href="https://www.bioinformatics.org/sewer/">SeWeR</a> (SEquence analysis using WEb Resources) (The Bioinformatics Organization, MBIdeas Innovation Center, U.S.A.) - is also a java suite for analyzing nucleic acids and proteins.</li>
      <br>

      <li><a href="http://workbench.sdsc.edu/">The Biology WorkBench</a> (San Diego Supercomputing Center, Univ. California San Diego, U.S.A.).  No longer funded.</li>
      <br>

      <li><a href="https://www.bioinformatics.org/sms2/">Sequence Manipulation Suite</a> or <a href="https://www.bioinformatics.org/sms/">here</a> (Bioinformatics Org).</li>
      <br>

      <li><a href="https://scratch.proteomics.ics.uci.edu/">SCRATCH Protein Predictor</a> (Institute for Genomics and Bioinformatics, University of California, Irvine, USA) - includes ACCpro: Solvent Accessibility;  ACCpro20: Solvent Accessibility (20 Class; SSpro: Secondary Structure (3 Class; SSpro8: Secondary Structure (8 Class); DISpro: Disorder; CONpro: Contact Number; CMAPpro: Contact Map; SVMcon: New SVM Contact Map; 3Dpro: Tertiary Structure; DOMpro: Domains;  DIpro: Disulfide Bonds; SOLpro: Solubility upon Overexpression; and, COBEpro: Continuous B-cell Epitopes.</li>
      <br>
      
      <li>
        <a href="https://bioinf.cs.ucl.ac.uk/psipred/?disopred=1">PSIPRED Protein Sequence Analysis Workbench</a> - includes PSIPRED v3.3 (Predict Secondary Structure); DISOPRED3 & DISOPRED2 (Disorder Prediction); pGenTHREADER (Profile Based Fold Recognition); MEMSAT3 & MEMSAT-SVM (Membrane Helix Prediction); BioSerf v2.0 (Automated Homology Modelling); DomPred (Protein Domain Prediction); FFPred 3 (Eukaryotic Function Prediction); GenTHREADER (Rapid Fold Recognition); MEMPACK (SVM Prediction of TM Topology and Helix Packing)   pDomTHREADER (Fold Domain Recognition); and, DomSerf v2.0 (Automated Domain Modelling by Homology). 
        <br>(<strong>Reference:</strong> Buchan DWA et al. 2013.  Nucl. Acids Res.  41 (W1): W340-W348).   
      </li>
      <br>

      <li><a href="http://rtools.cbrc.jp/cgi-bin/index.cgi">Rtools</a> (Computational Biology Research Consortium , Japan) - is a bioinformatics web Server for RNA (single FASTA format, <= 400nt) - In order to visualize the whole picture of the distribution of the secondary structure, this web-server provides users with rich information of single RNA sequences using 7 tools: (a) CentroidFold based on a generalized centroid estimator is one of the most accurate tools for predicting RNA secondary structures. (b) CentroidHomfold predicts RNA secondary structures by employing automatically collected homologous sequences of the target. (c) IPknot predicts RNA secondary structures including a wide class of pseudoknots. (d) Rchange computes entropy and internal energy changes of secondary structures for single-point mutated sequences. (e) CapR calculates probabilities that each RNA base position is located within each secondary structural context for long RNA sequences. (f) Raccess computes the accessibility of segment [a, b] = [x, x+l-1] in the transcript for all the positions x with fixed length l (Acc.len) = 5, 10, 20.  (g) RintD validates RNA secondary structures. Target secondary structures are predicted by CentroidFold (inference engine: McCaskill) and RNAfold (Minimum free energy structure).</li>
      <br>
    </ul>

    </section>

    <p class="updated"><strong>Updated:</strong> November, 2025</p>
  </main>

  <footer>
    <p>© 2025 Internet Resources for Molecular Biologists</p>
  </footer>

<script>
  const pages = [
    { name: "Bioinformatic Tutorials", link: "bioinformatics.html" },
    { name: "Carbohydrates", link: "carbohydrates.html" },
    { name: "Buffers", link: "buffers.html" },
    { name: "Metabolomics", link: "metabolomics.html" },
    { name: "Genomics", link: "genomics.html" },
    { name: "DNA Sequence Analysis", link: "https://google.com" },
    { name: "Protein Sequence Analysis", link: "https://google.com" },
    { name: "RNA Sequence Analysis", link: "https://google.com" },
    { name: "Transcription Analysis", link: "https://google.com" },
    { name: "Miscellaneous", link: "https://google.com" },
    { name: "Software", link: "https://google.com" },
    { name: "GenBank", link: "https://google.com" }
  ];

  const searchBar = document.getElementById("search-bar");
  const resultsDiv = document.getElementById("search-results");

  searchBar.addEventListener("input", searchSite);

  document.addEventListener("click", (e) => {
    if (!e.target.closest(".search-container")) {
      resultsDiv.innerHTML = "";
    }
  });

  function searchSite() {
    const input = searchBar.value.toLowerCase().trim();
    resultsDiv.innerHTML = "";
    if (input.length === 0) return;

    const filtered = pages.filter(page => page.name.toLowerCase().includes(input));
    if (filtered.length > 0) {
      filtered.forEach(page => {
        const link = document.createElement("a");
        link.href = page.link;
        link.textContent = page.name;
        link.target = "_self";
        resultsDiv.appendChild(link);
      });
    } else {
      resultsDiv.innerHTML = "<p>No matches found.</p>";
    }
  }
</script>

</body>
</html>
